pom.xml

```diff
@@ -65,10 +65,67 @@
     </dependencyManagement>
 
     <dependencies>
+        <dependency>
+            <groupId>io.jenkins.plugins</groupId>
+            <artifactId>javax-activation-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.jenkins.plugins</groupId>
+            <artifactId>jersey2-api</artifactId>
+            <version>2.35-6</version> <!-- TODO https://github.com/jenkinsci/bom/pull/1011 -->
+        </dependency>
         <dependency>
             <groupId>org.gitlab4j</groupId>
             <artifactId>gitlab4j-api</artifactId>
-	    <version>4.14.20</version>
+            <version>5.0.1</version>
+            <exclusions>
+                <!-- Provided by javax-activation-api plugin -->
+                <exclusion>
+                    <groupId>jakarta.activation</groupId>
+                    <artifactId>jakarta.activation-api</artifactId>
+                </exclusion>
+                <!-- Provided by core -->
+                <exclusion>
+                    <groupId>jakarta.servlet</groupId>
+                    <artifactId>jakarta.servlet-api</artifactId>
+                </exclusion>
+                <!-- Provided by jersey2-api plugin -->
+                <exclusion>
+                    <groupId>org.glassfish.jersey.connectors</groupId>
+                    <artifactId>jersey-apache-connector</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.glassfish.jersey.core</groupId>
+                    <artifactId>jersey-client</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.glassfish.jersey.inject</groupId>
+                    <artifactId>jersey-hk2</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.glassfish.jersey.media</groupId>
+                    <artifactId>jersey-media-json-jackson</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.glassfish.jersey.media</groupId>
+                    <artifactId>jersey-media-multipart</artifactId>
+                </exclusion>
+                <!-- TODO Unneeded test library https://github.com/gitlab4j/gitlab4j-api/pull/840 -->
+                <exclusion>
+                    <groupId>uk.org.webcompere</groupId>
+                    <artifactId>system-stubs-jupiter</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <!-- Needed by jersey-apache-connector, but jersey2-api plugin declares it as optional -->
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>apache-httpcomponents-client-4-api</artifactId>
+        </dependency>
```

Comment:
Should this be fixed in jersey2-api and make it non optional? again `apache-httpcomponents-client-4-api` is pretty in all Jenkins installs anyhow.

Reply:
We could. The downside is that someone who wants to use some other parts of Jersey 2 but doesn't need the Jersey Apache connector would be forced to install it. That might not be so bad, after all. I could go either way, don't feel strongly.

Reply:
Me neither, I was simply wondering so 🤷‍♂️